### ISSUE TITLE
Impossible to add a new variable without loosing all references

### ISSUE THREAD

**Issue by XavierVilars** [tokens-studio/figma-plugin#2055]
**Describe the bug**
- I have a list of variables I created using the "create variables" feature. 
- I recreated one by one all the references I need with other variable contained in a specific Figma file. 
- I created a new token in token studio. 
- To add it to my list of variable I hit the button "create variables" again.
- all my previous modifications are lost. No more references....

It's clearly a bloker.

**To Reproduce**
See above

**Expected behavior**
I want to be able to add, remove, edit tokens in Token Studio without loosing my previous work. Of course all is related to the fact that for the moment Token Studio is not in capacity to memorise the location of my tokens in my figma"s files.
Finally it's only asking for Token Studio to take into account a new parameter: the figma reference.

Cheers.

**Comment by six7** [comment]
We already support Figma references.

As long as
- the reference is a hard one (e.g. just `{colors.gray.500}`)
- the reference exists as a variable
- the referenced variable has a connection to a theme

we will apply the correct reference

This means, that if for example your `theme` set refers to a token from the `core` set, that you need to create a theme where your core tokens are created as a variable. That means having a structure like this: 



![Image](https://github.com/tokens-studio/figma-plugin/assets/4548309/e77b41b0-74b9-46b7-ac2f-2653cd8f41ea)


which was created by a theme structure like this:



![Image](https://github.com/tokens-studio/figma-plugin/assets/4548309/fdcf275b-898f-4fd9-ba11-452e192717dd)



and results in this



![Image](https://github.com/tokens-studio/figma-plugin/assets/4548309/f3bad10d-8bdc-4271-893b-bcedc64888a2)

**Comment by six7** [comment]
Closing, feel free to reopen if you feel it's not solved

**Comment by XavierVilars** [comment]
Hello Six7,
So I don't understand your answer :). I'm following headless structure. To do so I have 3 Figma files: 

- Core
- Semantic
- Component.

Semantic use Core references and components Semantic references. Components file don't know anything about core file. So for each file my list of variable is:
- Core file : Core theme / no references
- Semantic file: light theme, dark theme / references to Core file variables list.
- Components file: Tier 3 tokens / references to Semantic file variables list.

For usability reasons I don't want to duplicates all tokens as variables in all my files. Praticlay speaking if a designer use the semantic library in Figma, I don't want to expose Core variables. So yes I'm able to set to "don't publish" each variable I don't want to publish but sorry I don't want to do that on 200 variables one by one. Also if all my tokens are duplicated in multiple files in Figma then it means I do a modification in Core token then I need to be sure this mofication is applied in 3 differents files. It's a source of issues.

Tell me, may be I'm wrong ? :)

**Comment by XavierVilars** [comment]
my themes on Token studio:
![image](https://github.com/tokens-studio/figma-plugin/assets/104128778/156e386e-0ede-4932-a36a-17ba87b3c640)

**Comment by XavierVilars** [comment]
Don't know how to reopen it.

**Comment by esthercheran** [comment]
@XavierVilars I think #1974 and #2023 will solve your use case
